samples/minimum_criteria.rst

```diff
@@ -0,0 +1,98 @@
+.. _definition_and_minimum_criteria:
+
+Sample Definition and Minimum Criteria
+######################################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem,
+or module. For example, Kernel services, drivers, protocols, etc.
+Samples should be limited in scope and should focus only on demonstrating non-trivial or
+essential aspects of the subsystem(s) or module(s) being highlighted.
+
+Minimum Sample Criteria
+=======================
+
+1. Samples must not be used as a testcase.
+++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added.
```

Comment:
uhm, in the past I added samples also for cases for which a test was not really feasible or very hard to achieve.

Take for example the IPC service. In this case the sample is running on two different cores that are exchanging same data over shared memory. Having a test for this case is feasible with QEMU but requires a lot of additional work and external tooling.

So I propose to relax this a bit.

Reply:
Maybe you should not be talking about testcases being mandatory or talk about criteria for tests here, this is a subject for some other chapter. So I would remove this bullet completely to avoid consuion. Somewhere else here it can be mentioned that samples are optional.

Reply:
@carlocaione Have updated this to avoid confusion.